src/java.base/share/classes/java/text/NumberFormat.java

```diff
@@ -58,103 +58,91 @@
+ * independent of the locale conventions for decimal points, thousands-separators,
+ * whether the number format is even decimal, or even the particular decimal
+ * digits used. For example, this class could be used within an application to
+ * produce a number in a currency format according to conventions of the locale
+ * a user resides in.
  *
- * <p>
- * {@code NumberFormat} helps you to format and parse numbers for any locale.
- * Your code can be completely independent of the locale conventions for
- * decimal points, thousands-separators, or even the particular decimal
- * digits used, or whether the number format is even decimal.
+ * <h2>Getting a NumberFormat</h2>
+ * To get a {@code NumberFormat} for the default Locale, use one of the static
+ * factory methods that return a concrete subclass of {@code NumberFormat}.
+ * The following formats all provide an example of formatting the {@code Number}
+ * "2000.50" with the {@link java.util.Locale#US US} locale as the default locale.
+ * <ul>
+ * <li> Use {@link #getInstance()} or {@link #getNumberInstance()} to get
+ * a decimal format. For example, {@code "2,000.5"}.
+ * <li> Use {@link #getIntegerInstance()} to get an integer number format.
+ * For example, {@code "2,000"}.
+ * <li> Use {@link #getCurrencyInstance} to get a currency number format.
+ * For example, {@code "$2,000.50"}.
+ * <li> Use {@link #getCompactNumberInstance} to get a compact number format.
+ * For example, {@code "2K"}.
+ * <li> Use {@link #getPercentInstance} to get a format for displaying percentages.
+ * For example, {@code "200,050%"}.
+ * </ul>
  *
- * <p>
- * To format a number for the current Locale, use one of the factory
- * class methods:
- * <blockquote>
- * {@snippet lang=java :
- * myString = NumberFormat.getInstance().format(myNumber);
- * }
- * </blockquote>
- * If you are formatting multiple numbers, it is
- * more efficient to get the format and use it multiple times so that
- * the system doesn't have to fetch the information about the local
- * language and country conventions multiple times.
- * <blockquote>
- * {@snippet lang=java :
- * NumberFormat nf = NumberFormat.getInstance();
- * for (var myNumber : numbers) {
- *     output.println(nf.format(myNumber) + "; ");
- * }
- * }
- * </blockquote>
- * To format a number for a different Locale, specify it in the
- * call to {@code getInstance}.
- * <blockquote>
- * {@snippet lang=java :
- * NumberFormat nf = NumberFormat.getInstance(Locale.FRENCH);
- * }
- * </blockquote>
+ * Alternatively, if a {@code NumberFormat} for a different locale is required, use
+ * one of the factory method variants that take {@code locale} as a parameter,
+ * for example, {@link #getIntegerInstance(Locale)}. To determine if the current
+ * locale is supported by the installed locale-sensitive service implementation,
+ * either use {@link #getAvailableLocales()} or ensure a factory method call is enclosed
+ * within a try block.
```

Comment:
Not sure this is correct. Even if a locale is not "supported" by a locale provider, it won't throw an exception, but fallback to, say ROOT, locale.

Reply:
Good to know, I didn't realize this was the case. This is inherited from the original wording: `"This will work for the vast majority of locales; just remember to put it in a try block in case you encounter an unusual one."`

I'll replace it with the actual behavior, that Locale.ROOT is a fallback, if the particular locale is not supported.